service_test.go

```diff
@@ -143,427 +144,433 @@ var mockTransactions = []Nip47Transaction{
 }
 var mockTransaction = &mockTransactions[0]
 
-// TODO: split up into individual tests
-func TestHandleEvent(t *testing.T) {
+func TestBackwardsCompatibility(t *testing.T) {
+	// Test without adding a single permission
+}
+
+func TestHandleGetBalanceEvent(t *testing.T) {
 	ctx := context.TODO()
-	svc, _ := createTestService(t)
 	defer os.Remove(testDB)
-	//test not yet receivedEOS
-	res, err := svc.HandleEvent(ctx, &nostr.Event{
-		Kind: NIP_47_REQUEST_KIND,
-	})
-	assert.Nil(t, res)
-	assert.Nil(t, err)
-	//now signal that we are ready to receive events
-	svc.ReceivedEOS = true
-
-	senderPrivkey := nostr.GeneratePrivateKey()
-	senderPubkey, err := nostr.GetPublicKey(senderPrivkey)
+	mockLn, err := NewMockLn()
 	assert.NoError(t, err)
-	//test lnbc.. payload without having an app registered
-	ss, err := nip04.ComputeSharedSecret(svc.cfg.IdentityPubkey, senderPrivkey)
+	svc, err := createTestService(mockLn)
 	assert.NoError(t, err)
-	payload, err := nip04.Encrypt(nip47PayJson, ss)
+	_, app, ss, err := createUserWithApp(svc)
+	assert.NoError(t, err)
+
+	request := &Nip47Request{}
+	err = json.Unmarshal([]byte(nip47GetBalanceJson), request)
 	assert.NoError(t, err)
-	res, err = svc.HandleEvent(ctx, &nostr.Event{
-		ID:      "test_event_1",
+
+	// without permission
+	payload, err := nip04.Encrypt(nip47GetBalanceJson, ss)
+	assert.NoError(t, err)
+	reqEvent := &nostr.Event{
 		Kind:    NIP_47_REQUEST_KIND,
-		PubKey:  senderPubkey,
+		PubKey:  app.NostrPubkey,
 		Content: payload,
-	})
+	}
+
+	reqEvent.ID = "test_get_balance_without_permission"
+	res, err := svc.HandleGetBalanceEvent(ctx, request, reqEvent, *app, ss)
```

Comment:
I don't think the individual handle_x events should deal with encrypting the response. Then we can test encryption/decryption separately in one place. But maybe this could be added as a TODO?

Reply:
Yes, we can just return the `Nip47Response`, that would also DRY it up and we can avoid sending `ss` for no reason to those handlers